tests/core/pyspec/eth2spec/test/altair/transition/test_activation.py

```diff
@@ -0,0 +1,228 @@
+
+    assert spec.get_current_epoch(state) < fork_epoch
+    assert len(queuing_indices) > 0
+    for validator_index in queuing_indices:
+        assert not spec.is_active_validator(state.validators[validator_index], spec.get_current_epoch(state))
+
+    yield "pre", state
+
+    # irregular state transition to handle fork:
+    blocks = []
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch)
+    blocks.append(post_tag(block))
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    yield "blocks", blocks
+    yield "post", state
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=2)
+def test_transition_with_deposit_at_fork(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Create a deposit at the transition
+    """
+    transition_until_fork(spec, state, fork_epoch)
+
+    yield "pre", state
+
+    # create a new deposit
+    validator_index = len(state.validators)
+    amount = post_spec.MAX_EFFECTIVE_BALANCE
+    deposit = prepare_state_and_deposit(post_spec, state, validator_index, amount, signed=True)
+    deposit_old = prepare_state_and_deposit(spec, state, validator_index, amount, signed=True)
+    # sanity check: deposit operation is independent of spec fork versions
+    assert deposit_old == deposit
+    operation_dict = {'deposits': [deposit]}
+    # irregular state transition to handle fork:
+    state, block = do_altair_fork(state, spec, post_spec, fork_epoch, operation_dict=operation_dict)
+    blocks = []
+    blocks.append(post_tag(block))
+
+    assert not post_spec.is_active_validator(state.validators[validator_index], post_spec.get_current_epoch(state))
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    # finalize activation_eligibility_epoch
+    _, blocks_in_epoch, state = next_slots_with_attestations(
+        post_spec,
+        state,
+        spec.SLOTS_PER_EPOCH * 2,
+        fill_cur_epoch=True,
+        fill_prev_epoch=True,
+    )
+    blocks.extend([pre_tag(block) for block in blocks_in_epoch])
+    assert state.finalized_checkpoint.epoch == state.validators[validator_index].activation_eligibility_epoch
+
+    # continue regular state transition with new spec into next epoch
+    transition_to_next_epoch_and_append_blocks(post_spec, state, post_tag, blocks)
+
+    assert state.validators[validator_index].activation_epoch < post_spec.FAR_FUTURE_EPOCH
+
+    to_slot = state.validators[validator_index].activation_epoch * post_spec.SLOTS_PER_EPOCH
+    blocks.extend([
+        post_tag(block) for block in
+        state_transition_across_slots(post_spec, state, to_slot)
+    ])
+    assert post_spec.is_active_validator(state.validators[validator_index], post_spec.get_current_epoch(state))
```

Comment:
can consider skipping all of these blocks by just transitioning to `to_slot` and then perhaps making 1 block to "cap off the chain" but all of the epoch processing functions required will be run w/o making blocks

Reply:
I added `block_filter=only_at(to_slot)` to `state_transition_across_slots` arguments. Is that what you meant?